File: app/src/main/java/io/mainframe/hacs/main/BackDoorStatus.java
package io.mainframe.hacs.main;

public enum BackDoorStatus {

    UNKNOWN("-1"),
    OPEN("0"),
    CLOSED("1");

    private final String mqttValue;

    BackDoorStatus(String mqttValue) {
        this.mqttValue = mqttValue;
    }

    public static BackDoorStatus byMqttValue(String value) {
        for (BackDoorStatus status : values()) {
            if (status.getMqttValue().equals(value)) {
                return status;
            }
        }

        throw new IllegalArgumentException("Unsupported mqtt value: " + value);
    }

    public String getMqttValue() {
        return this.mqttValue;
    }
}

File: app/src/main/java/io/mainframe/hacs/main/BackDoorStatus.kt
package io.mainframe.hacs.main

enum class BackDoorStatus(val mqttValue: String) {
    UNKNOWN("-1"),
    OPEN("0"),
    CLOSED("1");

    companion object {
        fun byMqttValue(value: String): BackDoorStatus {
            for (status in entries) {
                if (status.mqttValue == value) {
                    return status
                }
            }

            throw IllegalArgumentException("Unsupported mqtt value: $value")
        }
    }
}
